tests/v2/validation/rbac/rbac.go

```diff
@@ -257,3 +257,69 @@ func createRole(client *rancher.Client, context string, roleName string, rules [
 	return
 
 }
+
+func editPsactCluster(client *rancher.Client, clustername string, namespace string, psact string) (clusterType string, err error) {
+	clusterID, err := clusters.GetClusterIDByName(client, clustername)
+	if err != nil {
+		return "", err
+	}
+	//Check if the downstream cluster is RKE2/K3S or RKE1
+	if strings.Contains(clusterID, "c-m-") {
+		clusterType = "RKE2K3S"
+		clusterObj, existingSteveAPIObj, err := clusters.GetProvisioningClusterByName(client, clustername, namespace)
+		if err != nil {
+			return "", err
+		}
+		oldPsact := clusterObj.Spec.DefaultPodSecurityAdmissionConfigurationTemplateName
+
+		for _, psactValue := range []string{psact, oldPsact} {
+			clusterObj.Spec.DefaultPodSecurityAdmissionConfigurationTemplateName = psactValue
+			_, err = clusters.UpdateK3SRKE2Cluster(client, existingSteveAPIObj, clusterObj)
+			if err != nil {
+				return clusterType, err
+			}
+			updatedClusterObj, _, err := clusters.GetProvisioningClusterByName(client, clustername, namespace)
+			if err != nil {
+				return "", err
+			}
+			if updatedClusterObj.Spec.DefaultPodSecurityAdmissionConfigurationTemplateName != psactValue {
+				errorMsg := "psact value was not reverted, Expected: " + psactValue + ", Actual: " + updatedClusterObj.Spec.DefaultPodSecurityAdmissionConfigurationTemplateName
+				return clusterType, errors.New(errorMsg)
+			}
+		}
+	} else {
+		clusterType = "RKE"
+		if psact == "" {
+			psact = " "
+		}
+		existingCluster, err := client.Management.Cluster.ByID(clusterID)
+		if err != nil {
+			return "", err
+		}
+		oldPsact := existingCluster.DefaultPodSecurityAdmissionConfigurationTemplateName
+
+		for _, psactValue := range []string{psact, oldPsact} {
+			updatedCluster := &management.Cluster{
+				Name: existingCluster.Name,
+				DefaultPodSecurityAdmissionConfigurationTemplateName: psactValue,
+			}
+			_, err = client.Management.Cluster.Update(existingCluster, updatedCluster)
+			if err != nil {
+				return clusterType, err
+			}
+			clusters.WaitForActiveRKE1Cluster(client, clusterID)
+			modifiedCluster, err := client.Management.Cluster.ByID(clusterID)
+			if err != nil {
+				return "", err
+			}
+			if psactValue == " " {
+				psactValue = ""
+			}
+			if modifiedCluster.DefaultPodSecurityAdmissionConfigurationTemplateName != psactValue {
```

Comment:
I included this check here and not the rbac_psa_test.go file because we are restoring the value to its original state within this function.

Reply:
Not sure I understand "If we move it to the rbac_psa_test.go file, we would need to verify if the value matches the original value" . Isn't that the check that we would be doing as part of the validation? Correct me if I'm getting it wrong.

I agree that the main objective of this test is to determine whether the psact value is updatable or not. However, this validation allows us to accurately assess whether the psact has been altered or remains unaffected.. 

 I assume we have other psact tests where we check psact specific behavior? -->. we currently do not have validations specifically for cluster changes. Validations in psact_rbac_test.go primarily focus on the psact value within the namespace rather than broader cluster-related modifications.

Reply:
Added additional checks in the rbac_psa_test.go file to create deployment after modifying the psact value.